lib/Sema/CSSimplify.cpp

```diff
@@ -7377,6 +7521,20 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyConformsToConstraint(
   if (type->isTypeVariableOrMember())
     return formUnsolved();
 
+  // ConformsTo constraints are generated when opening a generic
+  // signature with a RequirementKind::Conformance requirement, so
+  // we must handle pack types on the left by splitting up into
+  // smaller constraints.
+  if (auto *packType = type->getAs<PackType>()) {
+    for (unsigned i = 0, e = packType->getNumElements(); i < e; ++i) {
+      addConstraint(ConstraintKind::ConformsTo, packType->getElementType(i),
+                    protocol->getDeclaredInterfaceType(),
+                    locator.withPathElement(LocatorPathElt::PackElement(i)));
```

Comment:
I think you can move this to the first overload and remove duplicate logic from `matchExistentialTypes`.